src/main/java/org/isf/admission/manager/AdmissionBrowserManager.java

```diff
@@ -404,13 +417,27 @@ else if (dateOut.isBefore(ad.getAdmDate()) || dateOut.equals(ad.getAdmDate())) {
 									DateTimeFormatter.ISO_LOCAL_DATE.format(invalidEnd))));
 				}
 			}
-
+			List<Disease> diseaseIpdOuts1 = diseaseManager.getDiseaseIpdOut().stream().filter(dis-> dis == admission.getDiseaseOut1()).collect(Collectors.toList());
+			List<Disease> diseaseIpdOuts2 = diseaseManager.getDiseaseIpdOut().stream().filter(dis-> dis == admission.getDiseaseOut2()).collect(Collectors.toList());
+			List<Disease> diseaseIpdOuts3 = diseaseManager.getDiseaseIpdOut().stream().filter(dis-> dis == admission.getDiseaseOut3()).collect(Collectors.toList());
 			if (admission.getDiseaseOut1() == null && admission.getDisDate() != null) {
 				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.pleaseselectatleastfirstdiagnosisout.msg")));
 			} else if (admission.getDiseaseOut1() != null && admission.getDisDate() == null) {
 				errors.add(new OHExceptionMessage(MessageBundle.getMessage("angal.admission.pleaseinsertadischargedate.msg")));
+				throw new OHDataValidationException(errors);
```

Comment:
Why throw the exception?  That means subsequent tests are not done.  And if this is want you want to do (that is not find everything in one pass) then why isn't there a `throw new ....` after line 424 above?

Reply:
it's because it was like this before I only focused on the cases I added.

Reply:
And you broke the code.  If an error is found and the `throw` happens like you added no other checks are made and a collection object of `errors` would not be needed.   And there would be no need for the not is empty check of the errors collection at the end of this method.   Please remove the `throws` you have added.

Reply:
okay